components/scream/src/dynamics/homme/atmosphere_dynamics.cpp

```diff
@@ -161,156 +146,70 @@ set_updated_group (const FieldGroup<Real>& group)
-      using dp_type = std::remove_reference<decltype(dp)>::type;
-      dp = dp_type(dp_in.data(),num_elems);
-    } else if (name=="Q") {
-      // Tracers mass
-      auto& Q = tracers.Q;
-      auto Q_in = f.template get_reshaped_view<Scalar**[NP][NP][NVL]>();
-      using Q_type = std::remove_reference<decltype(Q)>::type;
-      Q = Q_type(Q_in.data(),num_elems,num_tracers);
-    } else {
-      ekat::error::runtime_abort("Error! Unexpected field name. This is an internal error. Please, contact developers.\n");
-    }
-  }
+  // Now that we have all fields set in homme, let's remap the input fields,
+  // so that Homme gets the correct Initial Conditions
+  m_p2d_remapper->remap(true);
 
-  // Required fields.
-  // NOTE: Homme's Elements store all field as views to non-const data. Why? I'm not
-  //       sure why we did that, but I assume it was to allow initialization.
-  //       Therefore, we need to cast away the const from the scream input fields.
-  // TODO: make Hommexx Elements structure store const views for stuff that is indeed (logically) const.
-  auto& forcing = Homme::Context::singleton().get<Homme::ElementsForcing>();
-  for (ci_string name : {"FQ", "FM", "FT", "ps"}) {
-    const auto& f = m_dyn_fields_in.at(name);
-
-    if (name=="FQ") {
-      // Tracers forcing
-      auto& fq = tracers.fq;
-      auto fq_in = f.template get_reshaped_view<const Scalar**[NP][NP][NVL]>();
-      using fq_type = std::remove_reference<decltype(fq)>::type;
-      auto non_const_ptr = const_cast<Scalar*>(fq_in.data());
-      fq = fq_type(non_const_ptr,num_elems,num_tracers);
-    } else if (name=="FM") {
-      // Momemntum forcing
-      auto& fm = forcing.m_fm;
-      // Use dynamic extent for second dimension, since preqx and theta have different extents
-      auto fm_in = f.template get_reshaped_view<const Scalar**[NP][NP][NVL]>();
-      using fm_type = std::remove_reference<decltype(fm)>::type;
-      auto non_const_ptr = const_cast<Scalar*>(fm_in.data());
-      fm = fm_type(non_const_ptr,num_elems);
-    } else if (name=="FT") {
-      // Temperature forcing
-      auto& ft = forcing.m_ft;
-      auto ft_in = f.template get_reshaped_view<const Scalar*[NP][NP][NVL]>();
-      using ft_type = std::remove_reference<decltype(ft)>::type;
-      auto non_const_ptr = const_cast<Scalar*>(ft_in.data());
-      ft = ft_type(non_const_ptr,num_elems);
-    } else if (name=="ps") {
-      // Surface pressure (use to init dp3d)
-      auto& ps = state.m_ps_v;
-      auto ps_in = f.template get_reshaped_view<const Real*[NTL][NP][NP]>();
-      using ps_type = std::remove_reference<decltype(ps)>::type;
-      auto non_const_ptr = const_cast<Real*>(ps_in.data());
-      ps = ps_type(non_const_ptr,num_elems);
-    } else {
-      ekat::error::runtime_abort("Error! Unexpected field name. This is an internal error. Please, contact developers.\n");
-    }
-  }
+  // Finish homme initialization
+  // Homme::initialize_dp3d_from_ps_c();
```

Comment:
Is this work still to be done?

Reply:
Ah, no. So, standalone Homme (from e3sm) does init dp3d from ps and from reference pressure levels thickness. However, in order to pick up ps from the initial conditions file, I would have had to expose ps as an input for HommeDynamics (otherwise it would not be given to Homme). Furthermore, even if I did that, `dp` would still need an I.C., according to the AD, since it's an input for the atm. Which would be confusing: you ask for an I.C. for `dp`, but it seems to not be refleted in the numbers (since the code will replace that with `dp=some_func(ps,ref_vert_coord)`.

I _think_ the easiest thing to do is to forget about `dp=some_func(ps,ref_vert_coord)`, and simply init `dp` with what's in the I.C. file. If you guys agree, I'll simply remove the commented line.

Reply:
That sounds okay to me. What do you think, @AaronDonahue ?

Reply:
I concur.  I'm guessing there will be some code cleanup in the future to address things like this.  For example if we decide it really doesn't make sense to include `dp` as an init'd variable.  But right now out goal is to get the code coupled and running.  So we can certainly revisit in the future if warranted.